app/javascript/app/components/sectors-agriculture/drivers-of-emissions/historical-emissions-graph/historical-emissions-graph-selectors.js

```diff
@@ -0,0 +1,298 @@
+    return {
+      axes: {
+        ...DEFAULT_AXES_CONFIG,
+        yLeft: { ...DEFAULT_AXES_CONFIG.yLeft, unit: 'MtCO2e' }
+      },
+      theme: colorThemeCache,
+      tooltip,
+      columns: {
+        x: [{ label: 'emission', value: 'x' }],
+        y: yColumns
+      }
+    };
+  }
+);
+
+const getFilterOptions = createSelector([getYColumns], yColumns => {
+  if (!yColumns) return null;
+  return yColumns.map(column => ({ ...column, groupId: 'subSectors' }));
+});
+
+/** PIE-CHART SELECTORS */
+const getGhgEmissionsFilter = createSelector(
+  [getEmissionCountrySelected],
+  selectedCountry => {
+    if (!selectedCountry) return GHG_DEFAULT_FILTER;
+    return { ...GHG_DEFAULT_FILTER, location: selectedCountry.value };
+  }
+);
+
+export const getPieChartData = createSelector([getGhgEmissionsData], data => {
+  if (!data || !data.length) return null;
+
+  const sectorsLastYearEmission = data
+    .map(({ emissions, sector, location }) => {
+      const lastYearEmission = emissions[emissions.length - 1];
+      return { sector, location, ...lastYearEmission };
+    })
+    .filter(({ value }) => value > 0); // filter for negative emission for Forestry sector
+
+  if (!sectorsLastYearEmission) return null;
+  const totalEmission = sectorsLastYearEmission.reduce(
+    (acc, emission) => acc + emission.value,
+    0
+  );
+
+  const sectorEmissions = sectorsLastYearEmission.map(({ sector, value }) => ({
+    name: getColumnValue(sector).toLowerCase(),
+    value,
+    sector,
+    formattedValue: `${format('.2s')(value)}`,
+    formattedPercentage: `${format('.2f')(value * 100 / totalEmission)}%`,
+    percentageValue: value * 100 / totalEmission
+  }));
+
+  const agricultureRow = sectorEmissions.find(
+    ({ name }) => name === 'agriculture'
+  );
+  const sectorsEmissionsData = agricultureRow
+    ? [
+      agricultureRow,
+      ...sectorEmissions.filter(({ name }) => name !== 'agriculture')
+    ]
+    : sectorEmissions;
+  const { location, year } = sectorsLastYearEmission[0];
+
+  return {
+    year,
+    location,
+    emissionValue: agricultureRow && agricultureRow.formattedValue,
+    emissionPercentage: agricultureRow && agricultureRow.formattedPercentage,
+    data: sectorsEmissionsData
+  };
+});
+
+const getPieChartConfig = createSelector([getPieChartData], pieChartData => {
+  if (!pieChartData || !pieChartData.data || !pieChartData.data.length) {
+    return null;
+  }
+  const { data } = pieChartData;
+  const columns = data.map(({ sector }) => ({
+    label: sector,
+    value: getColumnValue(sector).toLowerCase()
+  }));
+
+  const tooltip = getTooltipConfig(columns);
+  let theme = getPieChartThemeConfig(columns, GREY_CHART_COLORS);
+  const agricultureTheme = {
+    ...theme.agriculture,
+    stroke: AGRICULTURE_COLOR
+  };
+  theme = { ...theme, agriculture: agricultureTheme };
+
+  const config = {
+    tooltip,
+    animation: true,
+    axes: {
+      yLeft: {
+        unit: 'MtCO2e',
+        label: '2014'
```

Comment:
these two probably shouldn't be hardcoded?

Reply:
In fact, I don't need axes object here, removed & done